src/sentry/integrations/slack/actions/notification.py

```diff
@@ -71,9 +79,391 @@ def __init__(self, *args: Any, **kwargs: Any) -> None:
+                "payload": json_blocks,
+            }
+            if self.get_option("channel"):
+                log_params["channel_name"] = self.get_option("channel")
+
+            lifecycle.add_extras(log_params)
+            record_lifecycle_termination_level(lifecycle, e)
+            return None
+
+    def _save_issue_alert_notification_message(
+        self,
+        data: NewIssueAlertNotificationMessage,
+    ) -> None:
+        """Save an issue alert notification message to the repository."""
+        try:
+            self._issue_repository.create_notification_message(data=data)
+        except NotificationMessageValidationError as err:
+            extra = data.__dict__ if data else None
+            _default_logger.info(
+                "Validation error for new issue alert notification message",
+                exc_info=err,
+                extra=extra,
+            )
+        except Exception:
+            # if there's an error trying to save a notification message, don't let that error block this flow
+            # we already log at the repository layer, no need to log again here
+            pass
+
+    def _save_notification_action_message(
+        self,
+        data: NewNotificationActionNotificationMessage,
+    ) -> None:
+        """Save a notification action message to the repository."""
+        try:
+            self._action_repository.create_notification_message(data=data)
+        except NotificationMessageValidationError as err:
+            extra = data.__dict__ if data else None
+            _default_logger.info(
+                "Validation error for new notification action message", exc_info=err, extra=extra
+            )
+        except Exception:
+            # if there's an error trying to save a notification message, don't let that error block this flow
+            # we already log at the repository layer, no need to log again here
+            pass
+
+    def _get_issue_alert_thread_ts(
+        self,
+        lifecycle: EventLifecycle,
+        rule_id: int | None,
+        rule_action_uuid: str | None,
+        event: GroupEvent,
+        open_period_start: datetime | None,
+        new_notification_message_object: NewIssueAlertNotificationMessage,
+    ) -> str | None:
+        """Find the thread in which to post an issue alert notification as a reply.
+
+        Return None to post the notification as a top-level message.
+        """
+        # We need to search by rule action uuid and rule id, so only search if they exist
+        if not (
+            rule_action_uuid
+            and rule_id
+            and OrganizationOption.objects.get_value(
+                organization=self.project.organization,
+                key="sentry:issue_alerts_thread_flag",
+                default=ISSUE_ALERTS_THREAD_DEFAULT,
+            )
+        ):
+            return None
+
+        try:
+            parent_notification_message = self._issue_repository.get_parent_notification_message(
```

Comment:
I think we can break this dependency on `self`, and potentially extract this function to a new class/module entirely.

Does it make sense to have something like a `SlackThreadUtils` class while we figure out the shape of this builder?

Reply:
On a similar note, could we make most of these helpers class methods potentially? It doesn't seem like there's much reliance at all on the class's internal data.